internal/infrastructure/price-feeder/feed.go

```diff
@@ -0,0 +1,56 @@
+package pricefeederinfra
```

Comment:
Are we sure we need to export the types declared here? If not needed, we can move to `priceFeed` and `market` and make all their fields not exported too.

Also, if the factory func is not used outside of this folder, we should make it internal: `newPriceFeed`